src/traces/violin/layout_attributes.js

```diff
@@ -0,0 +1,19 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var boxLayoutAttrs = require('../box/layout_attributes');
+
+// TODO update descriptions
+// ... or maybe just reuse box* attributes?
+module.exports = {
+    violinmode: boxLayoutAttrs.boxmode,
+    violingap: boxLayoutAttrs.boxgap,
+    violingroupgap: boxLayoutAttrs.boxgroupgap
```

Comment:
Important: Should `box` and `violin` the same _gap_, _groupgap_ and even _mode_ attributes?

In other words, should `violin` be thought as a different _mode_ for `box`?

Reply:
I could see users wanting to use the `gap` attributes on violins the same way they would with boxes. imo it seems pretty intuitive for these charts to behave the same way and have the same grouping attributes available.

Reply:
Violin-specific violinmode, violingroup and violingroupgap are implemented in https://github.com/plotly/plotly.js/pull/2116/commits/7769f203a5b7697f682b6c5df19bc65677b72827